scipy/spatial/transform/rotation.py

```diff
@@ -0,0 +1,874 @@
+              to the original frame.
+            - As the physical rotation of a vector being glued to the original
+              frame as it rotates. In this case the vector components are
+              expressed in the original frame before and after the rotation.
+
+        In terms of DCMs, this application is the same as
+        `self.as_dcm().dot(vectors)`.
+
+        Parameters
+        ----------
+        vectors : array_like, shape (3,) or (N, 3)
+            Each `vectors[i]` represents a vector in 3D space. A single vector
+            can either be specified with shape `(3, )` or `(1, 3)`. The number
+            of rotations and number of vectors given must follow standard numpy
+            broadcasting rules: either one of them equals unity or they both
+            equal each other.
+        inverse : boolean, optional
+            If `inverse` is `True` then the inverse of the rotation(s) is
+            applied to the input vectors. Default is `False`.
+
+        Returns
+        -------
+        outvecs : `numpy.ndarray`, shape (3,) or (N, 3)
+            Result of applying rotation on input vectors.
+            Shape depends on the following cases:
+
+                - If object contains a single rotation (as opposed to a stack
+                  with a single rotation) and a single vector is specified with
+                  shape `(3,)`, then `output` has shape `(3,)`.
+                - In all other cases, `output` has shape `(N, 3)`, where `N` is
+                  either the number of rotations or vectors.
+        """
+        vectors = np.asarray(vectors)
+        if vectors.ndim > 2 or vectors.shape[-1] != 3:
+            raise ValueError("Expected input of shape (3,) or (P, 3), "
+                             "got {}.".format(vectors.shape))
+
+        single_vector = False
+        if vectors.shape == (3,):
+            single_vector = True
+            vectors = vectors[None, :]
+
+        dcm = self.as_dcm()
+        if self._single:
+            dcm = dcm[None, :, :]
+
+        n_vectors = vectors.shape[0]
+        n_rotations = len(self)
+
+        if n_vectors != 1 and n_rotations != 1 and n_vectors != n_rotations:
+            raise ValueError("Expected equal numbers of rotations and vectors "
+                             ", or a single rotation, or a single vector, got "
+                             "{} rotations and {} vectors.".format(
+                                n_rotations, n_vectors))
+
+        if inverse:
+            result = np.einsum('ikj,ik->ij', dcm, vectors)
+        else:
+            result = np.einsum('ijk,ik->ij', dcm, vectors)
+
+        if self._single and single_vector:
+            return result[0]
+        else:
+            return result
+
+    def __mul__(self, other):
+        """Compose this rotation with the other.
+
+        If `p` and `q` are two rotations, then the composition of 'q followed
```

Comment:
Here we had some discussion about how to interpret the order of application.

Do you have any ideas on how to improve this description? I think the current description is fine, however at least I used to think about rotations composition as consecutive frame rotations (so p * q is "p followed by q"). Obviously tricky, not sure what we can do.

Reply:
I've seen both conventions. Not really sure how to improve the "followed by". However, I think the dcm expression in the next sentence clarifies things.